src/rpdk/core/project.py

```diff
@@ -1345,21 +1352,68 @@ def _generate_stack_template_files(self) -> None:
             with ct_file.open("r") as f:
                 json_data = json.load(f)
             resource_name = self.type_info[2]
-            stack_template_data = {
-                "Description": f"Template for {self.type_name}",
-                "Resources": {
-                    f"{resource_name}": {
-                        "Type": self.type_name,
-                        "Properties": self._replace_dynamic_values(
-                            json_data["CreateInputs"]
-                        ),
-                    }
-                },
-            }
-            stack_template_file_name = f"{CANARY_FILE_PREFIX}{count}_001.yaml"
-            stack_template_file_path = stack_template_folder / stack_template_file_name
-            with stack_template_file_path.open("w") as stack_template_file:
-                yaml.dump(stack_template_data, stack_template_file, indent=2)
+
+            self._save_stack_template_data(
+                resource_name,
+                count,
+                stack_template_folder,
+                self._replace_dynamic_values(
+                    json_data[CREATE_INPUTS_KEY],
+                ),
+                CANARY_FILE_CREATE_SUFFIX,
```

Comment:
could you please explain why this needs to be hardcoded? Not sure this is the best customer experience. Can this be something configurable by the customer?


Ideally canaries should not test every functionality of the resource, this is what integration tests are for. However, if we dynamically assign names `001/00*` then it might got too far as providers have no upper limit of how many inputs they can specify. Can we introduce a setting inside rpdk config file where providers can configure how many canaries they can generate? Sometime you need more than just 2 files

Reply:
Some customer experience optimizations will be proposed/reviewed in a separate PR